source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst

```diff
@@ -1235,6 +1235,122 @@ the pool's generation number then error handling MUST continue according to
 stale and the client MUST NOT update any topology state.
 (See `Why ignore errors based on CMAP's generation number?`_)
 
+Error handling pseudocode
+~~~~~~~~~~~~~~~~~~~~~~~~~
+
+Application operations can fail in various places, for example:
+
+- A network error, network timeout, or command error may occur while
+  establishing a new connection. Establishing a connection includes the
+  MongoDB handshake and completing authentication (if configured).
+- A network error or network timeout may occur while reading or writing to an
+  established connection.
+- A command error may be returned from the server.
+- A "writeConcernError" field may be included in the command response.
+
+Depending on the context, these errors may update update SDAM state by marking
+the server Unknown and may clear the server's connection pool. Some errors
+also require other side effects, like cancelling a check or requesting an
+immediate check. Drivers may use the following pseudocode to guide their
+implementation::
+
+    def handleError(error):
+        address = error.address
+        topologyVersion = error.topologyVersion
+
+        with client.lock:
+            # Ignore stale errors based on generation and topologyVersion.
+            if isStaleError(client.topologyDescription, error)
+                return
+
+            if isStateChangeError(error):
+                # Mark the server Unknown
+                unknown = new ServerDescription(type=Unknown, error=error, topologyVersion=topologyVersion)
+                onServerDescriptionChanged(unknown, connection pool for server)
```

Comment:
Is `onServerDescriptionChanged` supposed to mean 'emit the serverDescription change' event?

Reply:
Yes it means "process the server description". It's defined here: https://github.com/mongodb/specifications/blob/master/source/server-discovery-and-monitoring/server-discovery-and-monitoring.rst#replacing-the-topologydescription